id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/database/IdentifiersDao.scala

```diff
@@ -5,37 +5,117 @@ import javax.inject.Singleton
 import com.google.inject.Inject
 import com.twitter.inject.Logging
 import scalikejdbc._
+
+import uk.ac.wellcome.models.SourceIdentifier
 import uk.ac.wellcome.platform.idminter.model.{Identifier, IdentifiersTable}
 import uk.ac.wellcome.utils.GlobalExecutionContext.context
 
 import scala.concurrent.{Future, blocking}
 
+case class UnableToMintIdentifierException(message: String)
+    extends Exception(message)
+
 @Singleton
 class IdentifiersDao @Inject()(db: DB, identifiers: IdentifiersTable)
     extends Logging {
 
   implicit val session = AutoSession(db.settingsProvider)
 
-  def lookupMiroID(miroID: String,
-                   ontologyType: String = "Work"): Future[Option[Identifier]] =
+  /* An unidentified record from the transformer can give us a list of
+   * identifiers from the source systems, and an ontology type (e.g. "Work").
+   *
+   * This method looks for existing IDs that have matching ontology type and
+   * source identifiers.
+   */
+  def lookupID(sourceIdentifiers: List[SourceIdentifier],
+               ontologyType: String): Future[Option[Identifier]] = {
+
+    // TODO: This exception should be handled gracefully, not sent around the
+    // TryBackoff ad infinitum.
+    if (sourceIdentifiers.isEmpty) {
+      throw new UnableToMintIdentifierException(
+        "No source identifiers supplied!")
+    }
     Future {
       blocking {
-        info(s"About to search for MiroID $miroID in Identifiers")
+        info(
+          s"About to search for existing ID matching $identifiers and $ontologyType")
         val i = identifiers.i
-        withSQL {
+        val query = withSQL {
           select
             .from(identifiers as i)
             .where
+
+            // We always want to match the ontology type, and this field
+            // in SQL is never null.
             .eq(i.ontologyType, ontologyType)
-            .and
-            .eq(i.MiroID, miroID)
-        }.map(Identifier(i)).single.apply()
+
+            // Add conditions for matching on different source identifiers.
+            .map { sql: ConditionSQLBuilder[String] =>
+              addConditionForLookingUpID(
+                sql = sql,
+                sourceIdentifiers = sourceIdentifiers,
+                column = i.MiroID,
+                identifierScheme = "miro-image-number"
```

Comment:
reminder to self: you put in the `IdentifierSchemes` module for a reason, you should use it!